controllers/kafkamanagement/kafkauser_controller.go

```diff
@@ -297,6 +297,61 @@ func (r *KafkaUserReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 			continue
 		}
 
+		if clusterEvent == models.ClusterDeletingEvent {
+			patch = user.NewPatch()
+			delete(user.Status.ClustersEvents, clusterID)
+			err = r.Status().Patch(ctx, user, patch)
+			if err != nil {
+				l.Error(err, "Cannot patch Kafka user resource status",
+					"initial permissions", user.Spec.InitialPermissions,
+					"kafka user options", user.Spec.Options,
+					"kafka user metadata", user.ObjectMeta,
+				)
+				r.EventRecorder.Eventf(
+					user, models.Warning, models.PatchFailed,
+					"Resource status patch is failed. Reason: %v",
+					err,
+				)
+
+				return models.ReconcileRequeue, nil
+			}
+
+			l.Info("Kafka user has been deleted",
+				"initial permissions", user.Spec.InitialPermissions,
+				"kafka user options", user.Spec.Options,
+			)
+
+			r.EventRecorder.Eventf(
+				user, models.Normal, models.Deleted,
+				"Resource is deleted",
```

Comment:
Resource is deleted -> User is detached from cluster

Reply:
done